### DISCUSSION TITLE
Suggestion

### DISCUSSION THREAD

**Discussion by FifthDream** [SignalFlagZ/sd-webui-civbrowser#65] [Ideas]
First off, let me say i love this extension and use it daily. it's very useful! I thank you for your time creating it, and  updating it. It's very much appreciated. 

I have a suggestion. I'm not even sure it's possible or how much work it may take to implement, but is there any chance filters could be added to block models by keywords, and/or by authors' names? There are a few authors who repeatedly post the same content and models, and some specific subjects i'd love to be able to keep from showing up in searches, and having the ability to filter them out would be great. 

Thanks for taking the time to listen!

**Comment by SignalFlagZ** [comment]
Thanks for the new feature suggestions.

The blacklist feature is a good idea. It would be nice if Civitai's Content Controls could be used with the API. Unfortunately, Content Controls are not available with the API.
Uploaders filtering seems the easiest. Keyword filtering is not suitable because it is too vague. Tag filtering is also not suitable because there are no clear rules.
How to manage the blacklist within the web UI is likely to be a difficult problem. It's easy to hide uploaders by adding a simple uploaders string to Settings. It's practical for up to about 10 uploaders, but it becomes difficult to use for more than that. 
I'm going to try implementing an experimental blacklist feature.

**Reply by SignalFlagZ** [reply]
5340ec3f84650001d08c972097d60c51406a33ed
Add blacklist and favorite features to dev branch. Try it in the dev branch.
New text boxes has been added to Settings. Enter creator names, separated by commas.
Blacklist hides cards. In Favorite, draw ⭐️ on the card.

**Reply by FifthDream** [reply]
This is great! So helpful. Thank you so much! Really appreciate your work. ⭐️

**Reply by SignalFlagZ** [reply]
Merged into the default branch as v2.5. Use the default (mod) branch. If you find a bug, please report it to the Issues.

**Reply by FifthDream** [reply]
Will do! Thanks again.

**Reply by SignalFlagZ** [reply]
You may need to reconfigure due to variable name changes.
Thanks.
